public/model/parser.ts

```diff
@@ -288,9 +281,8 @@ export default class Parser {
 							builder.selectSplitter(id);
 							break;
 						case 'branch':
-							const adder = builder.branch();
 							const pipeDef = parameters as IPipeDefinition;
-							(adder as AddPipeSeg)(pipeDef);
+							builder = builder.branch(pipeDef);
```

Comment:
```suggestion
							builder.branch(pipeDef);
```
does the builder need to be reassigned after calling its functions?

Reply:
It doesn't *need* to be, because the same reference is maintained. But that assumption requires knowledge of the internal behaviour of builder, which parser shouldn't realy have.
Since parser returns the builder, I'd lean towards saying to take the new reference and use that (even if we happen to know it's the same because we wrote builder)